sympy/polys/appellseqs.py

```diff
@@ -0,0 +1,258 @@
+r"""
+Efficient functions for generating Appell sequences.
+
+An Appell sequence is a zero-indexed sequence of polynomials `p_i(x)`
+satisfying `p_{i+1}'(x)=(i+1)p_i(x)` for all `i`. This definition leads
+to the following iterative algorithm:
+
+.. math :: p_0(x) = c_0,\ p_i(x) = i \int_0^x p_{i-1}(t)\,dt + c_i
+
+The constant coefficients `c_i` are usually determined from the
+just-evaluated integral and `i`.
+
+Appell sequences satisfy the following identity from umbral calculus:
+
+.. math :: p_n(x+y) = \sum_{k=0}^n \binom{n}{k} p_k(x) y^{n-k}
+
+References
+==========
+
+.. [1] https://en.wikipedia.org/wiki/Appell_sequence
+.. [2] Peter Luschny, "An introduction to the Bernoulli function",
+       https://arxiv.org/abs/2009.06743
+"""
+from sympy.polys.densearith import dup_mul_ground, dup_sub_ground
+from sympy.polys.densetools import dup_eval, dup_integrate
+from sympy.polys.domains import ZZ, QQ
+from sympy.polys.polytools import named_poly
+from sympy.utilities import public
+
+def dup_bernoulli(n, K):
+    """Low-level implementation of Bernoulli polynomials."""
+    if n < 1:
+        return [1]
+    p = [1, K(-1,2)]
```

Comment:
When working with domain elements we should use `K.one` and `K.zero` instead of `1` and `0`.

Reply:
See e.g. https://docs.sympy.org/latest/modules/polys/domainsintro.html#domain-elements-vs-sympy-expressions